### PR TITLE
parser and translator changes

### PR DESCRIPTION
**Pull request by jldav1d**
**CHANGES:**
- for-to loop support for decrement and increment depending on step value
- fix parser issues concerning logic_value
- changed step value to only accept whole_literals
- changed two_d_param to accept a whole literal
- fix translation issue for charr data type

### CODE REVIEW

File: source/LexicalAnalyzer/prepare.py
Comment (remoo69) [inline source/LexicalAnalyzer/prepare.py:1]: bat pala to bawal jed

File: source/SyntaxAnalyzer/Parser.py [hunk 1]
```diff
@@ -1934,7 +1939,7 @@ def loop_body_statement(self):
         elif self.match("kung", True):
             self.enforce()
             self.match("(")
-            self.condition()
+            self.logic_value()
```
Comment (remoo69) [inline source/SyntaxAnalyzer/Parser.py:1942]: so no condition na? pero kala ko working naman

File: source/SyntaxAnalyzer/Parser.py [hunk 2]
```diff
@@ -2305,31 +2310,55 @@ def logic_value(self):
             self.Tree.initialize_new()
             if self.match("Sus", True):
                 self.logic_expr()
+                self.allow_logic_op = True
```
Comment (remoo69) [inline source/SyntaxAnalyzer/Parser.py:2313]: new flag ba to

**Review by remoo69 — COMMENTED** [review]
check comments

File: source/SyntaxAnalyzer/Parser.py (hunk 1 above)
Comment (jldav1d) [reply source/SyntaxAnalyzer/Parser.py:1942]: ay pwedeng condition parin yan, hindi ko lang naiba

File: source/SyntaxAnalyzer/Parser.py (hunk 2 above)
Comment (jldav1d) [reply source/SyntaxAnalyzer/Parser.py:2313]: yepp para sa l_val_withparen

File: source/LexicalAnalyzer/prepare.py
Comment (jldav1d) [reply source/LexicalAnalyzer/prepare.py:1]: ayy dapat nakauncomment yan

File: source/SyntaxAnalyzer/Parser.py (hunk 1 above)
Comment (remoo69) [reply source/SyntaxAnalyzer/Parser.py:1942]: so readd condition?

File: source/SyntaxAnalyzer/Parser.py (hunk 2 above)
Comment (remoo69) [reply source/SyntaxAnalyzer/Parser.py:2313]: icic

File: source/SyntaxAnalyzer/Parser.py (hunk 1 above)
Comment (remoo69) [reply source/SyntaxAnalyzer/Parser.py:1942]: *re-add

File: source/LexicalAnalyzer/prepare.py
Comment (remoo69) [reply source/LexicalAnalyzer/prepare.py:1]: oki

**Review by remoo69 — APPROVED** [review]

**Pull request merged**